nixos/modules/services/printing/lprint.nix

```diff
@@ -0,0 +1,514 @@
+      };
+      defaultPrintQuality = lib.mkOption {
+        type = types.oneOf [ types.enum [
+          "draft"
+          "normal"
+          "high"
+        ] types.str ];
+        default = "normal";
+        description = ''
+          The default print quality.
+        '';
+      };
+      defaultPrintScaling = lib.mkOption {
+        type = types.oneOf [ types.enum [
+          "auto"
+          "fit"
+          "fill"
+          "none"
+        ] types.str ];
+        default = "auto";
+        description = ''
+          The default print scaling.
+        '';
+      };
+      defaultPrintSpeed = lib.mkOption {
+        type = types.int;
+        default = 0;
+        description = ''
+          The default print speed.
+        '';
+      };
+      printerDarkness = lib.mkOption {
+        type = types.int;
+        default = 50;
+        description = ''
+          The printer darkness.
+        '';
+      };
+      printerResolution = lib.mkOption {
+        type = types.str;
+        description = ''
+          The printer resolution.
+        '';
+      };
+      doubleSided = lib.mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether the printer supports double-sided printing.
+        '';
+      };
+    } // papplConfigOptions;
+  };
+  lprintConfig = config.services.printing.lprint;
+in
+{
+  options = {
+    services.printing.lprint = {
+      enable = lib.mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to enable the lprint service.
+        '';
+      };
+
+      package = lib.mkPackageOption pkgs "lprint" {};
+
+      papplOptions = lib.mkOption {
+        type = globalPapplConfigType;
+        default = {
+          dnssdName = "LPrint";
+          contact = "";
+        };
+      };
+
+      printers = lib.mkOption {
+        type = types.attrsOf (types.submodule {
+          options = {
+            printerOptions = lib.mkOption {
+              type = printerConfigType;
+            };
+            papplOptions = lib.mkOption {
+              type = printerPapplConfigType;
+            };
+          };
+        });
+      };
+    };
+  };
+
+  # Implementation
+  config = lib.mkIf lprintConfig.enable {
+    services.avahi = {
+      enable = true;
+      publish = {
+        enable = true;
+        userServices = true;
+      };
+    };
+
+    systemd.services.lprint = {
+      description = "LPrint Service";
+      after = [ "network.target" "nss-lookup.target" ];
+      requires = [ "avahi-daemon.socket" ];
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig = {
+        Type = "simple";
+        ExecStart = "${pkgs.lprint}/bin/lprint server -o log-file=- -o log-level=info";
+        Restart = "on-failure";
+        RestartSec = 5;
+        User = "lprint";
+        Group = "lprint";
+      };
+    };
+
+    users.users.lprint = {
+      isSystemUser = true;
+      home = "/var/lib/lprint";
```

Comment:
a dataDir option would be nice

Reply:
The `lprint` user's home directory doesn't actually store anything, so it would be largely pointless.

Reply:
then don't set a home directory at all

Reply:
That's an option? I'll do that then.